sootup.java.bytecode/src/main/java/sootup/java/bytecode/interceptors/typeresolving/TypeResolver.java

```diff
@@ -95,14 +97,13 @@ public boolean resolve(@Nonnull Body.BodyBuilder builder) {
       }
     }
 
-    builder.setLocals(
-        builder.getLocals().stream()
-            .map(
-                local -> {
-                  Type type = minCastsTyping.getMap().getOrDefault(local, local.getType());
-                  return local.withType(type);
-                })
-            .collect(Collectors.toSet()));
+    locals.stream()
+        .forEach(
+            local -> {
+              Type type = minCastsTyping.getMap().getOrDefault(local, local.getType());
+              Local newLocal = local.withType(type);
+              builder.replaceLocal(local, newLocal);
```

Comment:
lets just replace when something changed
```suggestion
              Type type = minCastsTyping.getMap().getOrDefault(local, local.getType());
              if(type != local.getType()){
                      Local newLocal = local.withType(type);
                      builder.replaceLocal(local, newLocal);
              }
```